src/components/Agreement/AgreementDocument.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Box, Markdown, textStyle, useTheme, useLayout, GU } from '@aragon/ui'
+
+function AgreementDocument({ content }) {
+  const { layoutName } = useLayout()
+  const theme = useTheme()
+  const compactMode = layoutName === 'small'
+
+  return (
+    <Box padding={0}>
+      <Article theme={theme} compact={compactMode}>
+        <Markdown content={content} />
+      </Article>
+    </Box>
+  )
+}
+
+const Article = styled.article`
+  overflow-y: auto;
+  max-height: ${100 * GU}px;
+  padding: ${({ compact }) => `${compact ? 2 * GU : 7 * GU}px`};
```

Comment:
Internal padding on the article to allow scrollbar to sit nicely against box edge